lib/engine/game/g_1880/step/buy_train.rb

```diff
@@ -65,12 +65,27 @@ def setup
 
           def buyable_trains(entity)
             trains_to_buy = super
-            trains_to_buy.reject! { |t| t.name == '2P' } if entity.trains.find { |t| t.name == '2P' }
+            trains_to_buy.reject! { |t| t.name == '2P' } if owns_restored2?(entity)
+            trains_to_buy << restored2_train if can_buy_restored2?(entity)
```

Comment:
Why do you have to both reject and add?

Reply:
It's fun train logic. Currently the 2p shows up when a player has enough money to buy the cheapest non-2p train.
I need the 2p to show up when the player has enough to pay for it and be removed when they already own one. 
Is there a better way?

Reply:
Why isn't it showing up? I couldn't find where it was being removed. Can you point me at that code?

Reply:
It's a combo of:
2p not being the cheapest depot train (due to not counting as train for EMR)
and this code:
https://github.com/tobymao/18xx/blob/master/lib/engine/step/train.rb#L152

Reply:
this is the bug

Reply:
Why don't we set EBUY_DEPOT_TRAIN_MUST_BE_CHEAPEST to false and then in buyable_trains, reject 2p if the player's cash is less than the cost of a 2p (or they have sold a share)?

Reply:
You know if shares have been sold by `@corporations_sold` not being empty

Reply:
it seems convoluted to change EBUY_DEPOT_TRAIN_MUST_BE_CHEAPEST value to handle a non ebuy scenario and rely on the fact that this game only has one train on sale at the time from the depot and no rule about ebuy trains and cheapest to make this work. If you think it's a better approach i'll do it, but i don't see how it's an improvement over this